### PR TITLE
Adds type constants to SettingsStore and use them across the system

### PR DESCRIPTION
**Pull request by NiklasBr**
## Changes in this pull request  
Why constants? I was inspired by #14414 and also…

image

Will update this PR once the docs in #14480 has been merged into 11.x.

### CODE REVIEW

File: models/Tool/SettingsStore.php [hunk 1]
```diff
@@ -24,7 +24,17 @@
  */
 final class SettingsStore extends Model\AbstractModel
 {
-    protected static array $allowedTypes = ['bool', 'int', 'float', 'string'];
+    public const TYPE_BOOLEAN = 'bool';
+    public const TYPE_FLOAT = 'float';
+    public const TYPE_INTEGER = 'int';
+    public const TYPE_STRING = 'string';
+
+    protected static $allowedTypes = [
```
Comment (blankse) [inline models/Tool/SettingsStore.php]: Don't remove type hint
```suggestion
    protected static array $allowedTypes = [
```

Comment (NiklasBr) [reply models/Tool/SettingsStore.php]: Thanks, didn't notice I did so.

File: models/Tool/SettingsStore.php [hunk 2]
```diff
@@ -24,7 +24,17 @@
...
+        self::TYPE_BOOLEAN,
+        self::TYPE_FLOAT,
+        self::TYPE_INTEGER,
+        self::TYPE_STRING,
+    ];
```
Comment (jdreesen) [inline models/Tool/SettingsStore.php]: Is there a reason this is not an enum?

File: models/Tool/SettingsStore.php (hunk 1 above)
Comment (jdreesen) [reply models/Tool/SettingsStore.php]: If we don't change this to an enum: I think this should be a `const` instead.

File: models/Tool/SettingsStore.php [hunk 3]
```diff
@@ -24,7 +24,17 @@
  */
 final class SettingsStore extends Model\AbstractModel
 {
-    protected static array $allowedTypes = ['bool', 'int', 'float', 'string'];
+    public const TYPE_BOOLEAN = 'bool';
+    public const TYPE_FLOAT = 'float';
+    public const TYPE_INTEGER = 'int';
+    public const TYPE_STRING = 'string';
+
+    protected static array $allowedTypes = [
```
Comment (jdreesen) [inline models/Tool/SettingsStore.php]: If we don't change this to an enum: I think this should be a `const` instead.

File: models/Tool/SettingsStore.php (hunk 2 above)
Comment (NiklasBr) [reply models/Tool/SettingsStore.php]: I think this is less of a BC break for everyone still using the string type variables.

Comment (jdreesen) [reply models/Tool/SettingsStore.php]: That's true. But we could deprecate it in Pimcore `10.6`.

**Comment by blankse** [comment]
We should also change the examples in the docs.

**Comment by lukmzig** [comment]
Hi @NiklasBr, thank you very much for this improvement. Two small things:
1. Could you please resolve the merge conflict?
2. SettingsStore::set method is used also in the tests, more specifically [here](https://github.com/pimcore/pimcore/blob/11.x/tests/Model/Tool/SettingsStoreTest.php). I think it would be good to replace type here with the constants which you introduced. Could you have a look?

Thank you!

**Comment by lukmzig** [comment]
Thank you @NiklasBr, looks good

**Pull request merged**